README.md

```diff
@@ -5,10 +5,90 @@
 [![tests](https://github.com/FFY00/mesonpy/actions/workflows/tests.yml/badge.svg)](https://github.com/FFY00/mesonpy/actions/workflows/tests.yml)
 [![codecov](https://codecov.io/gh/FFY00/mesonpy/branch/main/graph/badge.svg?token=xcb2u2YvVk)](https://codecov.io/gh/FFY00/mesonpy)
 
-Meson PEP 517 Python build backend.
+Python build backend ([PEP 517](https://www.python.org/dev/peps/pep-0517/)) for Meson projects.
 
-It works on both pure Python and compiled projects, and has *optional*
-[PEP 621](https://www.python.org/dev/peps/pep-0621/) metadata support.
+### Usage
+
+#### Enabling the build backend
+
+To use this build backend, you must specify it in your `pyproject.toml` file.
+
+```toml
+[build-system]
+build-backend = 'mesonpy'
+requires = [
+  'meson-python',
+]
+```
+
+If you have any other build dependencies, you must also add them to the
+`requires` list.
+
+#### Automatic metadata
+
+If no other metadata is specified, `mesonpy` will fetch the project name and
+version from Meson. In which case, you don't need to add anything else to your
+`pyproject.toml` file.
+
+This is not recommended. Please consider specifying the Python package metadata.
```

Comment:
I assume because this is metadata that belongs in `pyproject.toml` anyway? And for `mesonpy` itself it doesn't matter? If so, proposed rephrase: _"This works but is not recommended, because complete metadata in `pyproject.toml` has value for other tools."_

Reply:
Also, I suggest moving this section below the `Specifying metadata` one. Always best to start with the right way of doing things in user-facing docs.